aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -180,6 +180,26 @@ static TensorIterator make_reduction(
   return TensorIterator::reduce_op(viewed_result1, viewed_result2, self.to(dtype));
 }
 
+Tensor logcumsumexp(const Tensor& self, int64_t dim) {
+  auto result = at::empty(self.sizes(), self.options());
+  at::logcumsumexp_out(result, self, dim);
+  return result;
+}
+
+Tensor& logcumsumexp_out(Tensor& result, const Tensor& self, int64_t dim) {
+  check_scalar_type_device_layout_equal(result, self);
+  {
+    NoNamesGuard guard;
+    result.resize_(self.sizes());
+    auto cummax_values = std::get<0>(at::cummax(self, dim));
+    result = at::exp(self - cummax_values);
```

Comment:
setting the result in this way isn't correct, you need to write into the tensor.